airbyte-ci/connectors/base_images/base_images/python/v1.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+"""This module declares all the airbyte python connector base image for version 1.
+Please create a v2.py module if you want to declare a new major version.
+"""
+from __future__ import annotations
+
+from typing import Final, final
+
+import dagger
+from base_images import common, sanity_checks
+from base_images.python.common import AirbytePythonConnectorBaseImage, PythonBase
+
+
+class _1_0_0(AirbytePythonConnectorBaseImage):
+    base_base_image: Final[PythonBase] = PythonBase.PYTHON_3_9_18
+
+    changelog_entry: Final[
```

Comment:
@alafanechere I get the spirit of this! However this as a variable feels like an awkward fit
1. docstrings are not often a named variable
2. It adds a compilation step to our changelog
3. And this information gets lost if we delete this class from our codebase
    1. In the event we no longer use the dagger image, and its using methods that prevent us from updating to a new version of dagger
    1. We release a broken version 1.2.3, fix it in 1.2.4, I cant imagine we would want to keep a broken version around right?

This tells me that this info would better live in a changelog.md file (that we lint to ensure we are updating)

(an inbetween maybe moving to a docstring argument but still unsure on utility)

Reply:
> It adds a compilation step to our changelog

Why would it be problematic?
I like the idea of an changelog entry being added only if the "compile" step succeeds. Moreove in term of DX I like to have to modify a single file to make my change.

> And this information gets lost if we delete this class from our codebase
In the event we no longer use the dagger image, and its using methods that prevent us from updating to a new version of dagger
We release a broken version 1.2.3, fix it in 1.2.4, I cant imagine we would want to keep a broken version around right?

By design we don't want to delete a class. If we have to face this situation I'll change the changelog generation logic to not lose the entry.

> We release a broken version 1.2.3, fix it in 1.2.4, I cant imagine we would want to keep a broken version around right?

Actually I think we want to keep it around. I think we can make this package an analogy of a docker registry, if we release a broken version we won't delete it from our docker registry because we don't want to break downstream consumers. If a version is broken we release a new patch version and keep the broken one around.

Reply:
> However this as a variable feels like an awkward fit

But it's declared `Final` 😄 A `mypy` check will catch if anything mutates it.
Would you prefer a class decorator?

```python
@changelog_entry("Install new poetry version")
class _1_0_1(AirbytePythonConnectorBaseImage): 
```